### PR TITLE
Backend: Add Delete User Form

### PR DESCRIPTION
**Pull request by scouillard**
## Description

Controller method and Specs for the Delete User form.

### CODE REVIEW

**Comment by scouillard** [comment]
Depends on #3405.

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -21,6 +21,16 @@ def create
         end
       end
 
+      def destroy
+        user = User.find(params[:id])
+        if user.destroy
+          # TODO: samuel - Need to destroy all adjacent Rooms and Recordings as well
```
Comment (farhatahmad) [inline app/controllers/api/v1/users_controller.rb:27]: This is already done through the `dependent: :destroy`

File: spec/controllers/users_controller_specs.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  describe 'DELETE users#destroy' do
+    it 'deletes the user' do
+      user = create(:user)
+      delete :destroy, params: { id: user.id }
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'returns an error if the user id is invalid' do
+      delete :destroy, params: { id: 0 }
```
Comment (farhatahmad) [inline spec/controllers/users_controller_specs.rb:18]: 0 could be a valid id at some point. I would change it to nil, or something like `id: "invalid-id"`

**Comment by farhatahmad** [comment]
You'll need to rebase and resolve the conflicts

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -28,4 +28,18 @@
       expect(user.reload.name).to eq(user.name)
     end
   end
+
+  describe 'DELETE users#destroy' do
+    it 'deletes the user' do
+      user = create(:user)
+      delete :destroy, params: { id: user.id }
+      expect(response).to have_http_status(:ok)
+      expect { user.reload }.to raise_error(ActiveRecord::RecordNotFound)
```
Comment (farhatahmad) [inline spec/controllers/users_controller_spec.rb:37]: This works but isn't quite the approach we want to take. Take a look at some of the other tests to see how we do it

Comment (scouillard) [reply spec/controllers/users_controller_spec.rb:37]: Agreed - I'm sorry, I should have checked before.

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -28,4 +28,17 @@
...
+  describe 'DELETE users#destroy' do
+    it 'deletes the user' do
+      user = create(:user)
+      expect(response).to have_http_status(:ok)
+      expect { delete :destroy, params: { id: user.id } }.to change(User, :count).by(-1)
+    end
+
+    it 'returns an error if the user id is invalid' do
+      delete :destroy, params: { id: 'invalid-id' }
+      expect(response).to have_http_status(:not_found)
```
Comment (farhatahmad) [inline spec/controllers/users_controller_spec.rb:41]: This still doesn't actually check that something was deleted - just checks that a :not_found was returned.

We need to actually check that a record in the database was deleted

Comment (scouillard) [reply spec/controllers/users_controller_spec.rb:41]: The test at line 33 does that.
Not sure test at line 39 is any good - please let me know.

Comment (farhatahmad) [reply spec/controllers/users_controller_spec.rb:41]: Sorry meant to say **not** deleted. 

This should do exactly as 36, but instead check that it wasn't changed

**Pull request merged**